autoload/ale/c.vim

```diff
@@ -22,6 +23,102 @@ function! ale#c#FindProjectRoot(buffer) abort
     return ''
 endfunction
 
+function! ale#c#ParseCFlagsToList(path_prefix, cflags) abort
+    let l:previous_option = ''
+    let l:shell_option = 0
+    let l:macro_option = 0
+    let l:cflags_list = []
+
+    for l:option in a:cflags
+
+        " Check if cflag contained spaces
+        if l:shell_option || stridx(l:option, '=`') >= 0
+            " Cflag contained shell command with spaces (ex. -D='date +%s')
+            let l:shell_option = 1
+            let l:previous_option .= l:option
+            if l:option[-1: -1] isnot? '`'
+                let l:previous_option .= ' '
+                continue
+            endif
+            let l:shell_option = 0
+        elseif l:macro_option || stridx(l:option, '$((') > 0
+            " Cflag contained macro with spaces (ex -Da=$(( 4 * 20  )))
+            let l:macro_option = 1
+            let l:previous_option .= l:option
+            if stridx(l:option, '))') < 0
+                let l:previous_option .= ' '
+                continue
+            endif
+            let l:macro_option = 0
+        endif
+
+        if l:previous_option isnot? ''
+            let l:option = l:previous_option
+            let l:previous_option = ''
+        endif
+
+        " Fix relative paths if needed
+        if stridx(l:option, '-I') >= 0
+            if stridx(l:option, '-I' . s:sep) < 0
+                let l:option = '-I' . a:path_prefix . s:sep . l:option[2:]
```

Comment:
There is a bug here with escaping the path I think you should fix. I think `l:option[2;]` might already be escaped when returned by `make -n`, but your path prefix here won't be. You should probably write `ale#Escape(a:path_prefix) . ...`, in case the path contains spaces, etc.